test/utils/utils-test.js

```diff
@@ -68,4 +78,91 @@ describe('utils', () => {
       }, 1000)
     })
   })
+
+  describe('generateComponentsJson', () => {
+    const projectDir = path.join('build', 'componentsJSON')
+
+    const attachedExtensions = {
+      '@a/b': {
+        path: 'b'
+      },
+      '@a/c': {
+        path: 'c'
+      }
+    }
+
+    const extensionConfigs = {
+      '@a/b': {
+        components: [{
+          id: 'comp1',
+          type: 'type1',
+          path: 'path1'
+        }, {
+          id: 'comp2',
+          type: 'type2',
+          path: 'path2'
+        }]
+      },
+      '@a/c': {
+        components: [{
+          id: 'comp3',
+          type: 'type3',
+          path: 'path3'
+        }, {
+          id: 'comp4',
+          type: 'type4',
+          path: 'path4'
+        }]
+      }
+    }
+
+    const appSettings = {
+      loadAttachedExtensions: () => attachedExtensions,
+      getApplicationFolder: () => projectDir
+    }
+
+    const themes = [
+      'gmd',
+      'ios'
+    ]
+
+    beforeEach(async () => {
+      await fsEx.ensureDir(projectDir)
+
+      // ensure extensions and extension configs
+      for (let extensionId in attachedExtensions) {
+        const extDir = path.join(projectDir, EXTENSIONS_FOLDER, attachedExtensions[extensionId].path)
+        await fsEx.ensureDir(extDir)
+        await fsEx.writeJson(path.join(extDir, 'extension-config.json'), extensionConfigs[extensionId])
+      }
+
+      // ensure themes and theme config dirs
+      for (let i in themes) {
+        const themeDir = path.join(projectDir, THEMES_FOLDER, themes[i])
+        await fsEx.ensureDir(path.join(themeDir, 'config'))
+        await fsEx.writeJson(path.join(themeDir, 'extension-config.json'), {})
+      }
+    })
+
+    afterEach(async () => {
+      await fsEx.remove(projectDir)
+    })
+
+    it('should create components json', async () => {
+      utils.generateComponentsJson(appSettings)
+
+      const t1 = await fsEx.readJson(path.join(projectDir, THEMES_FOLDER, 'gmd', 'config', 'components.json'))
+      const t2 = await fsEx.readJson(path.join(projectDir, THEMES_FOLDER, 'ios', 'config', 'components.json'))
+
+      const result = {
```

Comment:
Would call this const "expected" or "expectedResult" maybe.

Reply:
done